### PR TITLE
Gracefully fail if test models weren't generated

### PR DESCRIPTION
**Pull request by vurusovs**
### Tickets:
 - CVS-109901

### CODE REVIEW

File: cmake/test_model_zoo.cmake [hunk 1]
```diff
@@ -8,7 +8,7 @@ if(ENABLE_OV_ONNX_FRONTEND)
     # if requirements are not installed automatically, we need to checks whether they are here
     ov_check_pip_packages(REQUIREMENTS_FILE "${OpenVINO_SOURCE_DIR}/src/frontends/onnx/tests/requirements.txt"
                           RESULT_VAR onnx_FOUND
-                          WARNING_MESSAGE "ONNX frontend tests will be skipped"
+                          WARNING_MESSAGE "ONNX testing models weren't generated, some tests will fail due .onnx models not found"
```
Comment (ilya-lavrenov) [inline cmake/test_model_zoo.cmake:11]: we can adjust message for PDPD and other places. Please, grep by `ov_check_pip_packages`

File: src/core/tests/pass/serialization/read_ir.hpp [hunk 2]
```diff
@@ -12,6 +12,7 @@ namespace ov {
 namespace test {
 
 inline std::shared_ptr<ov::Model> readModel(const std::string& model_path, const std::string& weights_path) {
+    OPENVINO_ASSERT(ov::util::file_exists(model_path), "Model ", model_path, " not found");
```
Comment (jane-intel) [inline src/core/tests/pass/serialization/read_ir.hpp:15]: You didn’t include it. Build failed

Comment (vurusovs) [reply src/core/tests/pass/serialization/read_ir.hpp:16]: Thanks a lot! Fixed

File: cmake/test_model_zoo.cmake (hunk 1 above)
Comment (vurusovs) [reply cmake/test_model_zoo.cmake:11]: Thanks! Done

File: src/core/tests/pass/serialization/read_ir.hpp [hunk 3]
```diff
@@ -27,7 +29,7 @@ inline std::shared_ptr<ov::Model> readModel(const std::string& model_path, const
     if (inputModel)
         return FE->convert(inputModel);
 
-    OPENVINO_ASSERT(false, "Failed to read the model");
+    OPENVINO_ASSERT(false, "Failed to read the model ", model_path);
```
Comment (ilyachur) [inline src/core/tests/pass/serialization/read_ir.hpp:32]:
```suggestion
    OPENVINO_THROW("Failed to read the model ", model_path);
```

File: src/core/tests/pass/serialization/read_ir.hpp [hunk 4]
```diff
@@ -46,7 +48,7 @@ inline std::shared_ptr<ov::Model> readModel(const std::string& model) {
     if (inputModel)
         return FE->convert(inputModel);
 
-    return nullptr;
+    OPENVINO_ASSERT(false, "Failed to read the model");
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply src/core/tests/pass/serialization/read_ir.hpp:51]: maybe we need to replace it globally in OpenVINO?

**Review by ilyachur — APPROVED** [review]
Please change to OPENVINO_THROW in the next PR

File: src/core/tests/pass/serialization/read_ir.hpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (vurusovs) [reply src/core/tests/pass/serialization/read_ir.hpp:51]: Done in https://github.com/openvinotoolkit/openvino/pull/19745

**Pull request merged**